rmrk_proxy/proxy.rs

```
@@ -0,0 +1,490 @@
+                ProxyError::BadMintValue
+            );
+
+            let total_assets = build_call::<DefaultEnvironment>()
+                .call(self.proxy.rmrk_contract.unwrap())
+                .gas_limit(GAS_LIMIT)
+                .exec_input(ExecutionInput::new(Selector::new(ink::selector_bytes!(
+                    "MultiAsset::total_assets"
+                ))))
+                .returns::<u32>()
+                .try_invoke()
+                .unwrap();
+            ensure!(total_assets.unwrap() > 0, ProxyError::NoAssetsDefined);
+            // This is temporary since current pseudo random generator is not working with big numbers.
+            ensure!(
+                total_assets.unwrap() <= MAX_ASSETS,
+                ProxyError::TooManyAssetsDefined
+            );
+
+            // TODO check why the call is failing silently when no or invalid transferred value is provided.
+            let mint_result = build_call::<DefaultEnvironment>()
+                .call(self.proxy.rmrk_contract.unwrap())
+                .gas_limit(GAS_LIMIT)
+                .transferred_value(transferred_value)
+                .exec_input(ExecutionInput::new(Selector::new(ink::selector_bytes!(
+                    "MintingLazy::mint"
+                ))))
+                .returns::<()>()
+                .try_invoke();
+            ink::env::debug_println!("mint_result: {:?}", mint_result);
+            mint_result
+                .map_err(|_| ProxyError::MintingError)?
+                .map_err(|_| ProxyError::MintingError)?;
+
+            let token_id = build_call::<DefaultEnvironment>()
+                .call(self.proxy.rmrk_contract.unwrap())
+                .gas_limit(GAS_LIMIT)
+                .exec_input(ExecutionInput::new(Selector::new(ink::selector_bytes!(
+                    "PSP34::total_supply"
+                ))))
+                .returns::<u64>()
+                .try_invoke()
+                .unwrap()
```

use match for result

---

Maybe you can return an error like [here](https://github.com/paritytech/ink/blob/8bb4b40d76adb28cf046f385cb5842bcd90b8787/integration-tests/multisig/lib.rs#L561)